spring-test/src/main/java/org/springframework/test/context/junit/jupiter/SpringExtension.java

```diff
@@ -159,7 +159,7 @@ private void validateAutowiredConfig(ExtensionContext context) {
 								testClass.getName(), Arrays.toString(methodsWithErrors)));
 			}, String.class);
 
-		if (errorMessage != NO_AUTOWIRED_VIOLATIONS_DETECTED) {
+		if (!NO_AUTOWIRED_VIOLATIONS_DETECTED.equals(errorMessage)) {
```

Comment:
FYI: the original code was intentional. Identity is sufficient here.

Please revert this change.

Reply:
Reverted [88d16fb](https://github.com/spring-projects/spring-framework/pull/29413/commits/88d16fb7730febd2f35e091d20674d81d30b5fa2)